crates/musli/src/serde/deserializer.rs

```diff
@@ -338,6 +340,11 @@ where
         self.decoder.skip()?;
         visitor.visit_unit()
     }
+
+    #[inline]
+    fn is_human_readable(&self) -> bool {
+        <<D::Cx as Context>::Mode as Mode>::is_human_readable()
```

Comment:
Until specialization lands which allows for default trait implementations for any mode type, we can implement this like this instead:

```rust
TypeId::of::<<D::Cx as Context::Mode>>() == TypeId::of::<crate::mode::Text>()
```

Reply:
When I try I'm getting this 

```rust

error[E0310]: the parameter type `D` may not live long enough
   --> crates/musli/src/serde/deserializer.rs:345:9
    |
345 |         TypeId::of::<<D::Cx as Context>::Mode>() == TypeId::of::<crate::mode::Text>()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         the parameter type `D` must be valid for the static lifetime...
    |         ...so that the type `D` will meet its required lifetime bounds
    |
help: consider adding an explicit lifetime bound
    |
37  |     D: Decoder<'de> + 'static,
    |                     +++++++++

```

Reply:
Interesting. Try setting the Mode associated type on the relevant traits to be `'static`. That should always be the case regardless.